Acrolinx.Sidebar/Util/Changetracking/DiffBasedLookup.cs

```diff
@@ -10,13 +10,42 @@
 
 namespace Acrolinx.Sdk.Sidebar.Util.Changetracking
 {
+    public enum DiffInputFormat
+    {
+        MARKUP,
+        TEXT
+    }
+    public class DiffOptions
+    {
+        public int diffTimeout { get; set; } = 5; // seconds
+        public int diffEditCost { get; set; } = 4;
+        public DiffInputFormat diffInputFormat { get; set; } = DiffInputFormat.TEXT;
+
+        internal short diffDualThreshold { get; } = 32;
+        internal float matchThreshold { get; } = 0.5f;
+        internal int matchDistance { get; } = 1000;
+        internal float patchDeleteThreshold { get; } = 0.5f;
+        internal short patchMargin { get; } = 4;
+        internal int matchMaxBits { get; } = 32;
+
+    }
+
     public class DiffBasedLookup
     {
         private string originalText;
+        private readonly DiffOptions diffOptions;
         public DiffBasedLookup(string originalText)
         {
             Contract.Requires(originalText != null);
             this.originalText = originalText;
+            diffOptions = new DiffOptions();
+        }
+
+        public DiffBasedLookup(string originalText, DiffOptions diffOptions)
```

Comment:
Diffoptions has only three values to set, why to pass whole object?

Reply:
Chnaged the options limited to three